packages/client/src/utils/hashes.ts

```diff
@@ -58,7 +58,9 @@ export async function hashes(
       const entry = map.get(h);
 
       if (entry) {
-        entry.names.push(name);
+        if (entry.names[0] !== name) {
```

Comment:
Is the conflicting entry always going to be at index 0? Why not use a Set?

Reply:
There's a lot of existing logic using `names[0]` so changing to a Set would complicate things.

https://github.com/vercel/vercel/blob/695f3a921220310a924c0af7dffed4eee306e060/packages/client/src/upload.ts#L88

I'm not even sure what the use case is for multiple names. Do you?

Reply:
I think it's so that you can have `foo.txt` and `bar.txt` that have the same file contents (same hash) but are two distinct files when re-populated in the build.

Reply:
Oh right, the name is the file path so there could be multiple files that has to the same contents.

I think its fine to leave it as-is since the duplication is only caused by my previous PR #7144 which has a second call to hashes()